main.rb

```diff
@@ -0,0 +1,114 @@
+# allows main.rb to access planet.rb and
+# solar_system.rb
+require_relative 'planet'
+require_relative 'solar_system'
+
+# main code that interacts with user
+# to create solar system and add planets
+def main
+  # let's user know what to expect
+  puts "Fill in the information below to create a planet within your solar system."
+
+  # user names their solar system so
+  # new solar system is created that can be
+  # accessed throughout the code
+  print "Name your solar system: "
+  solar_system_name = gets.chomp.to_s
+  solar_system_name = SolarSystem.new(solar_system_name)
+
+  # calls add_new_planet method to create new planet
+  add_new_planet(solar_system_name)
+
+  # default value is nil until user enters input
+  yes_or_no = nil
+  # until user doesn't want to add a new planet
+  # user will be asked to add new planet
+  until yes_or_no == "no"
+    # asks for user input
+    print "\nAdd another planet? (yes/no): "
+    yes_or_no = gets.chomp.to_s
+
+    # if user wants to add new planet
+    # user will be asked the same questions when
+    # they were asked to add the first planet
+    if yes_or_no =~ /[yY][eE][sS]/
+      add_new_planet(solar_system_name)
+    # if user doesn't want to add new planet
+    # they are given a few options to proceed
+    elsif yes_or_no =~ /[nN][oO]/
+      # default value of user_options is nil
+      user_options = nil
+
+      # until user wants to exit, options will
+      # continue to show up
+      until user_options == "3"
+        # asks user to make decision
+        print "\nOptions:\n1. List Planets\n2. Search Planet Details\n3. Exit\n"
```

Comment:
You've got a loop within a loop here! Often this is an indicator that you should break some part of this out as a separate method.